wcfsetup/install/files/lib/acp/form/CaptchaQuestionAddForm.class.php

```diff
@@ -3,198 +3,104 @@
-                'question' => I18nHandler::getInstance()->isPlainValue('question') ? I18nHandler::getInstance()->getValue('question') : '',
-            ]),
-        ]);
-        $returnValues = $this->objectAction->executeAction();
-        $questionID = $returnValues['returnValues']->questionID;
-
-        // set i18n values
-        $questionUpdates = [];
-        if (!I18nHandler::getInstance()->isPlainValue('question')) {
-            I18nHandler::getInstance()->save(
-                'question',
-                'wcf.captcha.question.question.question' . $questionID,
-                'wcf.captcha.question',
-                1
-            );
-
-            $questionUpdates['question'] = 'wcf.captcha.question.question.question' . $questionID;
-        }
-        if (!I18nHandler::getInstance()->isPlainValue('answers')) {
-            I18nHandler::getInstance()->save(
-                'answers',
-                'wcf.captcha.question.answers.question' . $questionID,
-                'wcf.captcha.question',
-                1
-            );
-
-            $questionUpdates['answers'] = 'wcf.captcha.question.answers.question' . $questionID;
-        }
-
-        if (!empty($questionUpdates)) {
-            $questionEditor = new CaptchaQuestionEditor($returnValues['returnValues']);
-            $questionEditor->update($questionUpdates);
-        }
-
-        $this->saved();
-
-        // reset values
-        I18nHandler::getInstance()->reset();
-        $this->isDisabled = 0;
-
-        // show success message
-        WCF::getTPL()->assign([
-            'success' => true,
-            'objectEditLink' => LinkHandler::getInstance()->getControllerLink(
-                CaptchaQuestionEditForm::class,
-                ['id' => $questionID]
-            ),
+        parent::createForm();
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    TextFormField::create('question')
+                        ->label('wcf.acp.captcha.question.question')
+                        ->maximumLength(255)
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.question.question\d+'),
+                    MultilineTextFormField::create('answers')
+                        ->label('wcf.acp.captcha.question.answers')
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.answers.question\d+')
+                        ->addValidator(new FormFieldValidator('regex', function (MultilineTextFormField $formField) {
+                            if (!empty($formField->getValidationErrors())) {
+                                return;
+                            }
+
+                            // returns `true` for a valid answer
+                            $answerValidator = function (string $text): bool {
```

Comment:
Should be `static` if `$this` is not referenced.
```suggestion
                            $answerValidator = static function (string $text): bool {
```